app/modules/mutations/publishModule.ts

```diff
@@ -3,148 +3,152 @@ import db from "db"
+export default resolver.pipe(
+  resolver.authorize(),
+  async ({ id, suffix }: { id: number; suffix?: string }) => {
+    const datetime = Date.now()
 
-  // TODO: Can be simplified along with stripe_webhook.ts
-  const module = await db.module.findFirst({
-    where: {
-      id,
-    },
-    include: {
-      license: true,
-      type: true,
-      authors: {
-        include: {
-          workspace: true,
-        },
+    // TODO: Can be simplified along with stripe_webhook.ts
+    const module = await db.module.findFirst({
+      where: {
+        id,
       },
-      references: {
-        include: {
-          authors: {
-            include: {
-              workspace: true,
-            },
-            orderBy: {
-              authorshipRank: "asc",
+      include: {
+        license: true,
+        type: true,
+        authors: {
+          include: {
+            workspace: true,
+          },
+        },
+        references: {
+          include: {
+            authors: {
+              include: {
+                workspace: true,
+              },
+              orderBy: {
+                authorshipRank: "asc",
+              },
             },
           },
         },
       },
-    },
-  })
+    })
 
-  if (!module!.main) throw Error("Main file is empty")
+    if (!module!.main) throw Error("Main file is empty")
 
-  const x = generateCrossRefObject({
-    schema: "5.3.1",
-    type: module!.type!.name,
-    title: module!.title,
-    language: module!.language,
-    authors: module!.authors!.map((author) => {
-      const js = {
-        firstName: author.workspace?.firstName,
-        lastName: author.workspace?.lastName,
-        orcid: author.workspace?.orcid,
-      }
+    const xmlData = generateCrossRefXml({
+      schema: "5.3.1",
+      type: module!.type!.name,
+      title: module!.title,
+      language: module!.language,
+      authors: module!.authors!.map((author) => {
+        const js = {
+          firstName: author.workspace?.firstName,
+          lastName: author.workspace?.lastName,
+          orcid: author.workspace?.orcid,
+        }
 
-      return js
-    }),
-    citations:
-      module!.references.length === 0
-        ? []
-        : module?.references.map((reference) => {
-            const refJs = {
-              publishedWhere: reference.publishedWhere,
-              authors:
-                reference.publishedWhere === "ResearchEquals"
-                  ? reference.authors.map((author) => {
-                      const authJs = {
-                        name: `${author.workspace?.firstName} ${author.workspace?.lastName}`,
-                        orcid: `https://orcid.org/${author!.workspace!.orcid}`,
-                      }
+        return js
+      }),
+      citations:
+        module!.references.length === 0
+          ? []
+          : module?.references.map(
+              ({ authors, authorsRaw, publishedAt, publishedWhere, suffix, prefix, title }) => {
```

Comment:
All this destructuring I'm doing is not super necessary, more of a personal preference that I maybe shouldn't do